### PR TITLE
Add Performance DB Table Construction Documentation

### PR DESCRIPTION
**Pull request by feba-rajan**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

This pull request introduces comprehensive documentation for the Performance Database, specifically focusing on how the tables are built, the base tables utilized.

### CODE REVIEW

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+>1. `endpoint_dataset_issue_type_summary`
+>2. `endpoint_dataset_resource_summary`
+>3. `endpoint_dataset_summary`
+>4. `provision_summary`
+>5. `reporting_historic_endpoints`
+>6. `reporting_latest_endpoints`
+
+### 1. Table: [endpoint_dataset_issue_type_summary](https://datasette.planning.data.gov.uk/performance/endpoint_dataset_issue_type_summary)
+
+**Purpose**: To summarize issues associated with each dataset and endpoint.
```
Comment (ssadhu-sl) [inline docs/data-operations-manual/Reference/Performance-db-tables-construction.md:16]: Maybe include something about the organisation. And say dataset and its associated endpoints

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+- `date`: Current date when the query is executed.
+- `issue_type`: Type of issues from the `issue_type` table.
+- `severity`: Severity of issues from the `issue_type` table.
+- `responsibility`: Responsibility assigned to the issues from the `issue_type` table.
+- `fields`: Concatenated fields from `issue` table.
+
+
+### 2. Table: [endpoint_dataset_resource_summary](https://datasette.planning.data.gov.uk/performance/endpoint_dataset_resource_summary)
+
+**Purpose**: To summarize resources associated with endpoints, including mapping and non-mapping fields.
```
Comment (ssadhu-sl) [inline docs/data-operations-manual/Reference/Performance-db-tables-construction.md:46]: Explain mapping and non-mapping fields

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+**Base Tables**:
+    - `provision`: Contains information about organizations, cohorts, and datasets sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
```
Comment (ssadhu-sl) [inline docs/data-operations-manual/Reference/Performance-db-tables-construction.md:49]: provision table -detail about datasets provisioned for a particular Organisation.

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+
+
+### 3. Table: [endpoint_dataset_summary](https://datasette.planning.data.gov.uk/performance/endpoint_dataset_summary)
+
+**Purpose**: To summarize endpoint information, including the latest statuses and exceptions.
+
+**Base Tables**:
+    - `endpoint`: Contains information about endpoints sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
+    - `source`: Provides a link between endpoints and organizations sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
```
Comment (ssadhu-sl) [inline docs/data-operations-manual/Reference/Performance-db-tables-construction.md:77]: Not sure if source means this, can you confirm from someone in DM?

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    - `log`: Contains logs related to endpoint performance sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
```
Comment (ssadhu-sl) [inline docs/data-operations-manual/Reference/Performance-db-tables-construction.md:78]: typo?

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+- `latest_status`: Extracted from the `log` table.
+- `latest_exception`: Extracted from the `log` table.
+- `entry_date`: Entry date of the endpoint.
+- `end_date`: End date of the endpoint.
+- `latest_resource_start_date`: Start date of the most recent resource.
+
+
+### 4. Table: [provision_summary](https://datasette.planning.data.gov.uk/performance/provision_summary)
+
+**Purpose**: To summarize provision metrics across organizations and datasets, counting various types of issues and endpoints.
```
Comment (ssadhu-sl) [inline docs/data-operations-manual/Reference/Performance-db-tables-construction.md:94]: explain what provision metrics means and type of issues might be confusing (user might think it like invalid org etc etc)

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+
+**Base Tables**:
+    - `provision`: Contains information about datasets sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
```
Comment (ssadhu-sl) [inline docs/data-operations-manual/Reference/Performance-db-tables-construction.md:97]: same as above.

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+    - `organisation`: Contains the names and details of organizations sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
+    - `issue`: Contains records of issues related to resources sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
+    - `resource`: Holds information about the resources linked to the endpoints sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
+    - `issue_type`: Defines types of issues sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
+    - `column_field`:  This is a mapping table that connects specific columns from an endpoint to their corresponding fields sourced according to the dataset's specification e.g UID -> reference, sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
+    - `endpoint`: Contains information about endpoints sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
+    - `source`: Provides a link between endpoints and organizations sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
```
Comment (ssadhu-sl) [inline docs/data-operations-manual/Reference/Performance-db-tables-construction.md:104]: same as above

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+    - `log`: Contains logs related to endpoint performance sourced from the [digital_land](https://datasette.planning.data.gov.uk/digital-land).
```
Comment (ssadhu-sl) [inline docs/data-operations-manual/Reference/Performance-db-tables-construction.md:105]: same as above

**Review by ssadhu-sl — CHANGES_REQUESTED** [review]
requested some changes

**Review by averheecke-tpx — APPROVED** [review]
lgtm

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md (hunk 9 above)
Comment (feba-rajan) [reply docs/data-operations-manual/Reference/Performance-db-tables-construction.md]: updated

File: docs/data-operations-manual/Reference/Performance-db-tables-construction.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+>1. `endpoint_dataset_issue_type_summary`
+>2. `endpoint_dataset_resource_summary`
+>3. `endpoint_dataset_summary`
+>4. `provision_summary`
+>5. `reporting_historic_endpoints`
+>6. `reporting_latest_endpoints`
+
+### 1. Table: [endpoint_dataset_issue_type_summary](https://datasette.planning.data.gov.uk/performance/endpoint_dataset_issue_type_summary)
+
+**Purpose**: To summarize issues associated with each dataset and its associated endpoint for various organisation.
```
Comment (greg-slater) [inline docs/data-operations-manual/Reference/Performance-db-tables-construction.md:16]: Would be useful to confirm here if there are any restrictions on resources. Does the table only include active resources, and does it include resources even when they don't have any issues?

Comment (feba-rajan) [reply docs/data-operations-manual/Reference/Performance-db-tables-construction.md:16]: - Includes only active resources(exclude where end_date = '')
- Excludes resources without issues(count_issue=0)

**Pull request merged**